src/fsharp/FSharp.Core/seq.fs

```diff
@@ -1468,13 +1473,13 @@ namespace Microsoft.FSharp.Collections
         let inline indexNotFound() = raise (new System.Collections.Generic.KeyNotFoundException(SR.GetString(SR.keyNotFoundAlt)))
         
         [<CompiledName("ToComposer")>]
-        let toComposer (source:seq<'T>): SeqComposer.SeqEnumerable<'T> = 
+        let toComposer (source:seq<'T>): Composer.Internal.SeqEnumerable<'T> = 
             checkNonNull "source" source
             match source with
-            | :? SeqComposer.Enumerable.EnumerableBase<'T> as s -> upcast s
-            | :? array<'T> as a -> upcast SeqComposer.Array.Enumerable((fun () -> a), SeqComposer.IdentityFactory.IdentityFactory)
-            | :? list<'T> as a -> upcast SeqComposer.List.Enumerable(a, SeqComposer.IdentityFactory.IdentityFactory)
-            | _ -> upcast SeqComposer.Enumerable.Enumerable<'T,'T>(source, SeqComposer.IdentityFactory.IdentityFactory)
+            | :? Composer.Enumerable.EnumerableBase<'T> as s -> upcast s
+            | :? array<'T> as a -> upcast Composer.Array.Enumerable((fun () -> a), Composer.IdentityFactory.IdentityFactory)
+            | :? list<'T> as a -> upcast Composer.List.Enumerable(a, Composer.IdentityFactory.IdentityFactory)
+            | _ -> upcast Composer.Enumerable.Enumerable<'T,'T>(source, Composer.IdentityFactory.IdentityFactory)
         
         let inline foreach f (source:seq<_>) =
```

Comment:
Would it be faster to write this `foreach` function as the one-liner `(toComposer source).ForEach f`? I don't know what the F# compiler does with the structure `|> fun x -> x.Method arg`: does it create a lambda, or does the lambda get optimized away? If the lambda is *not* optimized away, then I think `(toComposer source).ForEach f` might be a performance gain.